xrootd/cmd/xrd-srv/main_test.go

```diff
@@ -6,6 +6,7 @@ package main // import "go-hep.org/x/hep/xrootd/cmd/xrd-srv"
 
 import (
 	"context"
+	"crypto/rand"
```

Comment:
do we need `crypto/rand`?
couldn't we get away with just `math/rand` ?
(or is it to leverage the hardware acceleration?)

Reply:
>(or is it to leverage the hardware acceleration?)

yeah, `crypto/rand` is almost 4 times faster for me than `math/rand`:

```go

func BenchmarkCRand(b *testing.B) {
	bigData := make([]byte, 10*1024)
	for i := 0; i < b.N; i++ {
		_, err := crand.Read(bigData)
		if err != nil {
			b.Fatalf("could not prepare test data: %v", err)
		}
	}
}

func BenchmarkMRand(b *testing.B) {
	bigData := make([]byte, 10*1024)
	for i := 0; i < b.N; i++ {
		_, err := mrand.Read(bigData)
		if err != nil {
			b.Fatalf("could not prepare test data: %v", err)
		}
	}
}
```

```goos: windows
goarch: amd64
pkg: go-hep.org/x/hep/xrootd/cmd/xrd-srv
BenchmarkCRand-4   	  300000	      5870 ns/op
BenchmarkMRand-4   	  100000	     21472 ns/op
PASS
```

Reply:
following our interesting dicussion about `sync.Map` & `map`, I've just ran these benchmarks (with `go version devel +48c79734ff Fri Jul 20 20:08:15 2018 +0000 linux/amd64`) on my machine at the lab:

```
$> go test -v -bench . -benchmem 
goos: linux
goarch: amd64
pkg: bench-rand
BenchmarkCRand-8   	   30000	     41579 ns/op	       0 B/op	       0 allocs/op
BenchmarkERand-8   	  100000	     17370 ns/op	       0 B/op	       0 allocs/op
BenchmarkMRand-8   	  200000	     10951 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	bench-rand	5.882s
```

(where `ERand` is using `golang.org/x/exp/rand`)

I'd be tempted to go with `math/rand` :)

Reply:
here is my machine at the lab:
```
$> cat /proc/cpuinfo
processor	: 7
vendor_id	: GenuineIntel
cpu family	: 6
model		: 158
model name	: Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz
stepping	: 9
microcode	: 0x84
cpu MHz		: 800.038
cache size	: 8192 KB
physical id	: 0
siblings	: 8
core id		: 3
cpu cores	: 4
apicid		: 7
initial apicid	: 7
fpu		: yes
fpu_exception	: yes
cpuid level	: 22
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb invpcid_single pti ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm mpx rdseed adx smap clflushopt intel_pt xsaveopt xsavec xgetbv1 xsaves dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp
bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass
bogomips	: 7202.00
clflush size	: 64
cache_alignment	: 64
address sizes	: 39 bits physical, 48 bits virtual
power management:
```

IIRC, yours are: 
- `i3-6100 CPU @ 2.30GHz` (`windows`)
- `i5-7600 CPU @ 3.50GHz` (`linux`)

so... I don't know whether that means the `windows/amd64` compiler is making a very good job with leveraging hardware instructions for `crypto/rand` and a poor job at doing the same for `math/rand`, or if the `linux/amd64` compiler is doing a very poor job at leveraging h/w instructions for `crypto/rand`...

could you retry with also `exp/rand` on your windows machine?

(well, it's mostly out of curiosity at this point: I hope finding a file handle isn't on the hot-path.)

Reply:
Yeah, it's like this:
```
BenchmarkCRand-4          200000              6422 ns/op               0 B/op          0 allocs/op
BenchmarkMRand-4          100000             26688 ns/op               0 B/op          0 allocs/op
BenchmarkERand-4           30000             40259 ns/op               0 B/op          0 allocs/op
```